File: Exercism/javascript/scrabble-score/scrabble-score.js
'use strict'

const inverseKeyValues = (acc, curr) =>
  [...acc, ...curr[1].map(
    (char) => [char, curr[0]]
  )]

const SCORES = new Map([
  [1, ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T']],
  [2, ['D', 'G']],
  [3, ['B', 'C', 'M', 'P']],
  [4, ['F', 'H', 'V', 'W', 'Y']],
  [5, ['K']],
  [8, ['J', 'X']],
  [10, ['Q', 'Z']]
].reduce(inverseKeyValues, []))

const sum = (acc, curr) => acc + curr

export const score = (word) =>
  word === ''
    ? 0
    : word
      .toUpperCase()
      .split('')
      .map(char => SCORES.get(char))
      .reduce(sum)

File: Exercism/python/scrabble-score/scrabble_score.py
SCORES = dict()
for d in (dict.fromkeys('AEIOULNRST', 1),
          dict.fromkeys('DG', 2),
          dict.fromkeys('BCMP', 3),
          dict.fromkeys('FHVWY', 4),
          dict.fromkeys('K', 5),
          dict.fromkeys('JX', 8),
          dict.fromkeys('QZ', 10)):
    SCORES.update(d)


def score(word: str) -> int:
    """Computes the scrabble score for a given word."""
    return sum(SCORES[char] or 0 for char in word.upper())
